### PR TITLE
[red-knot] Improve various tracing logs

### PR DESCRIPTION
**Pull request by AlexWaygood**
Helps with #13011

### CODE REVIEW

File: crates/red_knot_python_semantic/src/types/infer.rs [hunk 1]
```diff
@@ -915,8 +923,16 @@ impl<'db> TypeInferenceBuilder<'db> {
     ///   - `from .foo import bar` => `tail == "foo"`
     ///   - `from ..foo.bar import baz` => `tail == "foo.bar"`
     fn relative_module_name(&self, tail: Option<&str>, level: NonZeroU32) -> Option<ModuleName> {
+        tracing::trace!(
+            "Attempting to resolve module '{}' relative to file '{}'",
+            Self::format_import_from_module(level.get(), tail),
```
Comment (MichaReiser) [inline crates/red_knot_python_semantic/src/types/infer.rs:928]: I would make this a standalone function. Makes it easier to call.

Comment (MichaReiser) [reply crates/red_knot_python_semantic/src/types/infer.rs:928]: I'm not if we want to both log attempting and the resolved module. It seems relatively verbose.

File: crates/red_knot_python_semantic/src/types/infer.rs [hunk 2]
```diff
@@ -955,12 +971,23 @@ impl<'db> TypeInferenceBuilder<'db> {
         // `follow_nonexistent_import_bare_to_module()`.
         let ast::StmtImportFrom { module, level, .. } = import_from;
         tracing::trace!("Resolving imported object {alias:?} from statement {import_from:?}");
+        let module = module.as_deref();
         let module_name = if let Some(level) = NonZeroU32::new(*level) {
-            self.relative_module_name(module.as_deref(), level)
+            tracing::debug!(
+                "Resolving imported object '{}' from module '{}' relative to file '{}'",
```
Comment (MichaReiser) [inline crates/red_knot_python_semantic/src/types/infer.rs:977]: Same here. We should avoid making debug too verbose. Maybe consider making it a trace or removing it entirely

**Review by MichaReiser — APPROVED** [review]
Thanks

File: crates/red_knot_python_semantic/src/types/infer.rs (hunk 2 above)
Comment (AlexWaygood) [reply crates/red_knot_python_semantic/src/types/infer.rs:965]: I made it a trace

**Pull request merged**